packages/babel-plugin-transform-typescript/test/fixtures/namespace/namespace-nested-module/input.ts

```diff
@@ -0,0 +1,6 @@
+namespace N {
+  const M1 = {}
+  export module M1 {}
+  const { M2 } = {}
+  export module M2 {}
```

Comment:
Babel currently throws for
```ts
const { M2 } = {}
export module M2 {}
```
because `M2` is never added to `names`, and `TSModuleDeclaration` handler will add `M2` variable declarations which collide with `const { M2 } = {}`.